multi-agent/tests/integration/blueprints/test_blueprint_versioning_regression.py

```
@@ -0,0 +1,384 @@
+
+        # The OCC $inc must have written version=2 onto the document.
+        raw = bp_col.find_one({"blueprint_id": bid})
+        assert raw["version"] == 2
+
+        # A snapshot at version=1 (the legacy state) must have been created.
+        snap = ver_repo.find_one(bid, 1)
+        assert snap is not None
+        assert snap.spec_dict_snapshot["name"] == "Legacy Blueprint"
+
+    def test_multiple_edits_succeed_after_pre_migration_first_edit(
+        self, service, identity, bp_col, ver_repo
```

Ruff (0.15.17)

[warning] Unused method argument: `identity`

(ARG002)